src/Compilers/CSharp/Portable/Symbols/TypeSymbol.cs

```diff
@@ -812,7 +812,34 @@ public ImmutableArray<SymbolDisplayPart> ToDisplayParts(CodeAnalysis.NullableFlo
 
         #region Interface member checks
 
-        protected SymbolAndDiagnostics FindImplementationForInterfaceMemberInNonInterfaceWithDiagnostics(Symbol interfaceMember)
+        /// <summary>
+        /// Locate implementation of the <paramref name="interfaceMember"/> in context of the current type.
+        /// The method is using cache to optimize subsequent calls for the same <paramref name="interfaceMember"/>.
+        /// </summary>
+        /// <param name="interfaceMember">Member for which an implementation should be found.</param>
+        /// <param name="ignoreImplementationInInterfacesIfResultIsNotReady">
+        /// The process of looking up an implementation for an accessor can involve figuring out how corresponding event/property is implemented,
+        /// <see cref="CheckForImplementationOfCorrespondingPropertyOrEvent"/>. And the process of looking up an implementation for a property can
+        /// involve figuring out how corresponding accessors are implemented, <see cref="FindMostSpecificImplementationInInterfaces"/>. This can 
+        /// lead to cycles, which could be avoided if we ignore the presence of implementations in interfaces for the purpose of
+        /// <see cref="CheckForImplementationOfCorrespondingPropertyOrEvent"/>. Fortunately, logic in it allows us to ignore the presence of
+        /// implementations in interfaces and we use that.
+        /// When the value of this parameter is true and the result that takes presence of implementations in interfaces into account is not
+        /// available from the cache, the lookup will be performed ignoring the presence of implementations in interfaces. Otherwise, result from
+        /// the cashe is returned.
+        /// When the value of the parameter is false, the result from the cache is returned, or calculated, taking presence of implementations
+        /// in interfaces into account and then cached.
+        /// This means that:
+        ///  - A symbol from an interface can still be returned even when <paramref name="ignoreImplementationInInterfacesIfResultIsNotReady"/> is true.
+        ///    A subsequent call with <paramref name="ignoreImplementationInInterfacesIfResultIsNotReady"/> false will return the same value. 
+        ///  - If symbol from a non-interface is returned for <paramref name="ignoreImplementationInInterfacesIfResultIsNotReady"/> is true. A subsequent
+        ///    call with <paramref name="ignoreImplementationInInterfacesIfResultIsNotReady"/> false will return the same value.
+        ///  - If no symbol is returned for <paramref name="ignoreImplementationInInterfacesIfResultIsNotReady"/> true. A subsequent call with
+        ///    <paramref name="ignoreImplementationInInterfacesIfResultIsNotReady"/> might return a symbol, but that symbol guaranteed to be from an interface.
+        ///  - If the first request is done with <paramref name="ignoreImplementationInInterfacesIfResultIsNotReady"/> true. A subsequent call
```

Comment:
>If the first request is done with <paramref name="ignoreImplementationInInterfacesIfResultIsNotReady"/> true. [](start = 15, length = 109)

I might be missing a subtlety, but it seems like the fourth bullet subsumes the first two bullets.

Reply:
> I might be missing a subtlety, but it seems like the fourth bullet subsumes the first two bullets.

I think there is a typo here it should say ```<paramref name="ignoreImplementationInInterfacesIfResultIsNotReady"/> false```. 

---
In reply to: [325392215](https://github.com/dotnet/roslyn/pull/38683#discussion_r325392215) [](ancestors = 325392215)